From 6d5903a35436fb71f4a245f7b9ddd6279d640adc Mon Sep 17 00:00:00 2001
From: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
Date: Fri, 15 Mar 2024 14:34:11 +0800
Subject: [PATCH 55/61] mtk: mac80211: fix mlo BW 160 channel switch issue

The link_id argument for cfg80211_ch_switch_started_notify is missing
after maintainer rebasing for chanreq
The original commit has link_id instead of 0
https://patchwork.kernel.org/project/linux-wireless/list/?series=821321&state=*

CR-Id: WCNCR00274293
Signed-off-by: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
---
 net/mac80211/cfg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/mac80211/cfg.c b/net/mac80211/cfg.c
index cca3e08..856c956 100644
--- a/net/mac80211/cfg.c
+++ b/net/mac80211/cfg.c
@@ -4045,7 +4045,7 @@ __ieee80211_channel_switch(struct wiphy *wiphy, struct net_device *dev,
 	}
 
 	cfg80211_ch_switch_started_notify(sdata->dev,
-					  &link_data->csa_chanreq.oper, 0,
+					  &link_data->csa_chanreq.oper, link_id,
 					  params->count, params->block_tx);
 
 	if (changed) {
-- 
2.39.2

